pkg/collector/corechecks/servicediscovery/impl_linux.go

```diff
@@ -228,15 +250,22 @@ func (li *linuxImpl) getServiceInfo(p proc, openPorts map[int]portlist.List) (*s
 	pInfo := processInfo{
 		PID:     p.PID(),
 		CmdLine: cmdline,
-		Env:     env,
-		Cwd:     cwd,
+		Env:     procResp.Proc.Environ,
+		Cwd:     procResp.Proc.CWD,
 		Stat: procStat{
 			StartTime: startTimeSecs,
 		},
 		Ports: ports,
 	}
 
 	meta := li.serviceDetector.Detect(pInfo)
+	if meta.Name == "" {
```

Comment:
We should make sure that name is almost always returned from service detection and that it is an error condition when that happens.

Reply:
That would change the existing api for the `usm` package to return `result, error` instead of `result, bool`. 
I suggest leaving this refactor for a future PR to keep the scope as small as possible for this one.